pkg/helm/chart.go

```diff
@@ -0,0 +1,275 @@
+// ChartLoader is the interface that wraps the Load method.
+type ChartLoader interface {
+	Load(chartURL string) (*chart.Chart, error)
+}
+
+// ChartManager satisfies the ChartLoader interface.
+type ChartManager struct {
+	httpGetter       HttpGetter
+	loadChartArchive func(io.Reader) (*chart.Chart, error)
+}
+
+// NewDefaultChartManager creates a new ChartManager with the default dependencies.
+func NewDefaultChartManager() *ChartManager {
+	return NewChartManager(
+		http.DefaultClient,
+		loader.LoadArchive,
+	)
+}
+
+// NewChartManager creates a new ChartManager with the explicit dependencies.
+func NewChartManager(
+	httpGetter HttpGetter,
+	loadChartArchive func(io.Reader) (*chart.Chart, error),
+) *ChartManager {
+	return &ChartManager{
+		httpGetter:       httpGetter,
+		loadChartArchive: loadChartArchive,
+	}
+}
+
+// Load loads a chart from a URL.
+// TODO(f0rmiga): implement caching for chart archives.
+func (cm *ChartManager) Load(chartURL string) (*chart.Chart, error) {
+	chartResp, err := cm.httpGetter.Get(chartURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load chart: %v", err)
+	}
+	defer chartResp.Body.Close()
+
+	chartRequested, err := cm.loadChartArchive(chartResp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load chart: %v", err)
+	}
+
+	return chartRequested, err
+}
+
+// ChartHelmClientProvider is the interface that wraps the methods for providing Helm action clients
+// for installing and uninstalling charts.
+type ChartHelmClientProvider interface {
+	ProvideInstaller(releaseName, namespace string) (HelmClientInstallRunner, error)
+	ProvideUninstaller(namespace string) (HelmClientUninstallRunner, error)
+}
+
+// ChartHelm satisfies the ChartHelmClientProvider interface.
+type ChartHelm struct {
+	configProvider     ConfigProvider
+	actionNewInstall   func(*action.Configuration) *action.Install
+	actionNewUninstall func(*action.Configuration) *action.Uninstall
+}
+
+// NewDefaultChartHelm creates a new ChartHelm with the default dependencies.
+func NewDefaultChartHelm() *ChartHelm {
+	return NewChartHelm(
+		NewDefaultConfig(),
+		action.NewInstall,
+		action.NewUninstall,
+	)
+}
+
+// NewChartHelm creates a new ChartHelm with the explicit dependencies.
+func NewChartHelm(
+	configProvider ConfigProvider,
+	actionNewInstall func(*action.Configuration) *action.Install,
+	actionNewUninstall func(*action.Configuration) *action.Uninstall,
+) *ChartHelm {
+	return &ChartHelm{
+		configProvider:     configProvider,
+		actionNewInstall:   actionNewInstall,
+		actionNewUninstall: actionNewUninstall,
+	}
+}
+
+// ProvideInstaller provides a Helm action client for installing charts.
+func (ch *ChartHelm) ProvideInstaller(
+	releaseName string,
+	namespace string,
+) (HelmClientInstallRunner, error) {
+	cfg, err := ch.configProvider.Provide(namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to provide chart installer: %v", err)
+	}
+	client := ch.actionNewInstall(cfg.(*action.Configuration))
```

Comment:
With the `cfg` becoming a pointer to `action.Configuration`, we won't need this type assertion here and in the `ProvideUninstaller` as well.